### PR TITLE
Refactor arg handling for da-lib.sh and docker-admin*

### PR DESCRIPTION
**Pull request by VictorLowther**
This does the following things:

* Pull identical argument handling in from docker-admin and
  docker-admin-up to da-lib.sh
* Always have the compose pieces clean up and recreate the .yml files
  when starting containers.

### CODE REVIEW

File: tools/da-lib.sh [hunk 1]
```diff
@@ -27,6 +27,62 @@ rebar() {
...
+args=()
+
+while (( $# > 0 )); do
+    arg="$1"
+    case $arg in
+        --clean) export COMPOSE_RESET="Y";;
+        --no-pull) export NO_PULL="Y";;
+        --access)
+             shift
+             containers["access"]="FORWARDER";;
```
Comment (galthaus) [inline tools/da-lib.sh]: This needs to take the next arg in the set.  It could be HOST or FORWARDER.

Comment (VictorLowther) [inline tools/da-lib.sh]: ok

File: tools/da-lib.sh [hunk 2]
```diff
@@ -27,6 +27,62 @@ rebar() {
     docker exec compose_rebar_api_1 rebar -E http://127.0.0.1:3000 -U rebar -P rebar1 "$@"
 }
 
+known_containers=(provisioner logging debug node access)
```
Comment (galthaus) [inline tools/da-lib.sh]: The OS_TYPE set of four lines needs to be moved after the argument processing and the COMPOSE_ACCESS needs to test the containers['access'] variable for "FORWARDER"

File: tools/da-lib.sh [hunk 3]
```diff
@@ -61,27 +117,24 @@ bring_up_admin_containers() {
...
     fi
 
-    if [ "$COMPOSE_RESET" == "Y" ] ; then
-        ./init_files.sh --clean
-    fi
-    if [[ ! -f docker-compose.yml ]] ; then
-        ./init_files.sh $COMPOSE_ARGS
-    fi
-    
+    ./init_files.sh --clean
```
Comment (galthaus) [inline tools/da-lib.sh]: This is good.  I thought I liked it originally, but it is so fast, it doesn't matter and works cleaner.

File: tools/da-lib.sh [hunk 4]
```diff
@@ -27,6 +27,62 @@ rebar() {
...
+    [[ ${containers["provisioner"]} ]] || containers["provisioner"]=true
+    [[ ${containers["access"]} ]] || containers["access"]=FORWARDER
+}
+
+args=()
+
+while (( $# > 0 )); do
+    arg="$1"
+    case $arg in
+        --clean) export COMPOSE_RESET="Y";;
```
Comment (galthaus) [inline tools/da-lib.sh]: This can be removed as well.

Comment (VictorLowther) [inline tools/da-lib.sh]: huh?  use_container is used elsewhere in the code, and docker_admin_default_containers is used by docker-admin and anything that expects a docker-admin like env.

Comment (galthaus) [inline tools/da-lib.sh]: I meant the --clean part.  :-)

**Pull request merged**